Menu

wisit-grouplease opened the pull request.

wisit-grouplease committed:
Create Project view and menubar

wisit-grouplease committed:
Update add view and create Product

nutteen commented on service/src/main/java/th/co/grouplease/simple/pm/service/project/read/model/ProjectEntry.java:
```diff
@@ -48,7 +48,7 @@
 
     @JsonIgnore
-    @ManyToOne(fetch = FetchType.LAZY)
+    @ManyToOne(fetch = FetchType.EAGER)
```
เปลี่ยนเป็น eager ทำไมครับ?

nutteen commented on service/src/main/java/th/co/grouplease/simple/pm/service/resource/ResourceController.java:
```diff
@@ -25,7 +25,7 @@
     @GetMapping
     public Page<Resource> getAllResources(@RequestParam @Min(value = 0, message = "Page must be at least 0") Integer page,
-                                          @RequestParam @Min(value = 1, message = "Page size must be at least 1") Integer pageSize){
+                                          @RequestParam @Min(value = 1, message = "Page size must be at least s1") Integer pageSize){
```
ระวังพิมพ์ผิด

nutteen commented on ui/src/main/java/th/co/grouplease/simple/pm/ui/component/PMMenuBar.java:
```diff
@@ -0,0 +1,21 @@
+        SubMenu m2Resource = m1Resource.getSubMenu();
+        MenuItem m2ResourceTeam = m2Resource.addItem("Resource Team");
+        MenuItem m2ResourceAll = m2Resource.addItem("Resource All");
+
```
เอา new line ออกครับให้ code มีบรรทัดน้อยที่สุด

nutteen commented on ui/src/main/java/th/co/grouplease/simple/pm/ui/main/ProjectView.java:
```diff
@@ -0,0 +1,110 @@
+
+import java.util.List;
+
+
```
เอา newline ออก

nutteen commented on ui/src/main/java/th/co/grouplease/simple/pm/ui/main/ProjectView.java:
```diff
@@ -0,0 +1,110 @@
+        ComboBox<Product> cbProducts = new ComboBox<>("Product");
+        cbProducts.setItemLabelGenerator(Product::getName);
+        List<Product> listOfProduct = productService.getProducts(0,productService.getProductCount());
+        cbProducts.setItems(listOfProduct);
```
ใช้ data provider แทนครับ

nutteen commented on ui/src/main/java/th/co/grouplease/simple/pm/ui/service/ProjectService.java:
```diff
@@ -0,0 +1,60 @@
+        return Math.toIntExact(restTemplate.getForObject("http://localhost:8080/projects/count", Long.class));
+    }
+
+    public Project createProduct(Project project) {
```
createProject ครับไม่ใช่ product

wisit-grouplease committed:
1 แก้ไข Combo เป็น DataProvider

2 แก้ไข api StartProjectCommand ใหใส่วันที่ startDate
3. เพิ่ม ContextMenu ใน Grid เพื่อทำ StartProjectCommand
